openhands/runtime/__init__.py

```diff
@@ -1,25 +1,79 @@
+from dataclasses import dataclass
+
 from openhands.runtime.e2b.sandbox import E2BBox
 
 
+@dataclass
+class RuntimeInfo:
+    module: str
+    class_name: str
+
+
+# Core runtimes as default
+_registered_runtimes = {
+    'eventstream': RuntimeInfo('client', 'EventStreamRuntime'),
+    'e2b': RuntimeInfo('e2b', 'E2bRuntime'),
+    'remote': RuntimeInfo('remote', 'RemoteRuntime'),
+}
```

Comment:
I think it might be clear to directly improve all these Runtime and maintain a map rather than doing dynamic import?

Import each runtime probably don't slow us down too much, so i'd actually prefer we go with a simpler/readable route (dynamic import can be hard to read). At least we can click the symbol and jump to source in the IDE - WDYT?

Reply:
Importing each will cause cyclic imports unless some good amount of refactoring is done elsewhere... I tried.
And I find this solution's readability pretty much the same as the original multi-if code.

Reply:
If that's the case, WDYT about this approach we are doing for agenthub?

We can maintain a registry on the base class
https://github.com/All-Hands-AI/OpenHands/blob/ee6a1cf3341beaba3b91705d86bcfe02d37f15da/openhands/controller/agent.py#L68-L81

and each new runtime just need to wrap itself to register:

https://github.com/All-Hands-AI/OpenHands/blob/ee6a1cf3341beaba3b91705d86bcfe02d37f15da/agenthub/codeact_agent/__init__.py#L4

Reply:
This way we don't need to explictly write out the mapping - we just need to clean up the `openhands/runtime` folder structure a bit

Reply:
I'll look into that approach.

Reply:
Every runtime implementation would have to inherit from BaseRuntime - 

Can we do smth like this (should get around the cyclic import?)

```
from .base import Runtime

class FancyRuntime(Runtime):
    def foo():
        pass

Runtime.register('fancy', FancyRuntime)
```

Reply:
As of now, the import of the runtime happens only when the `get_runtime_cls` is called e.g. during startup (idk why the `e2b` is an exception there),
because importing them all at once then when runtime is imported, it leads to cyclic import errors and the process aborts (e.g. make run).

What is the use case of a registry? Afaik we do not need access to multiple runtimes at the same time?
Is the issue here just "code style" over functionality? 🤔